src/agent/ncp_openthread.cpp

```diff
@@ -117,6 +121,12 @@ otbrError ControllerOpenThread::Init(void)
 
     mInstance = otSysInit(&mConfig);
     otCliUartInit(mInstance);
+
+#if defined(OPENTHREAD_POSIX_VENDOR_CLI_HEADER)
```

Comment:
Maybe this is not `#if'd` but instead we provide a weak definition of the function to be called. As it stands no the function would be `vendorCliInit` but may it should be renamed to `otVendorCliInit` or `otbrVendorCliInit`?

Reply:
I'd rather just leave this as a `#ifdef` because we're already already using a `#ifdef` above to include the header file. What advantages do you see for using the weakly-defined alias?

Reply:
Consistency with the way otPlatform APIs are done.  If on the header MACRO is a little awkward here too.

Reply:
This isn't an `otPlat` API though. The `#if` on the header macro isn't something completely out there though. This concept is used in the cli_uart.cpp file for `OT_CLI_UART_LOCK_HDR_FILE` too https://github.com/openthread/openthread/blob/master/src/cli/cli_uart.cpp#L60-L64

Maybe this is something better left up for discussion with the OT community?

Reply:
Actually, I'm fine with using a weakly defined function. Where would I put it though? I think it would have to be in the openthread repo and we would need to add a central place for the definitions. If we just threw it into [third_party/openthread/repo/src/cli/cli.cpp](https://github.com/openthread/openthread/blob/master/src/cli/cli.cpp#L4899), future `otVendor____()` functions might be defined in other files and finding them all would be a huge mess

Reply:
I still think the the macro used to optionally include this code should be something like `OPENTHREAD_ENABLE_OTBR_VENDOR_HOOK`

Reply:
Also, re `OPENTHREAD_POSIX_VENDOR_CLI_HEADER` I think what you are saying is that it maps back to this logic
```
target_compile_definitions(ot-vendor-cli PUBLIC
    OPENTHREAD_POSIX_VENDOR_CLI_HEADER="${VENDOR_CLI_HEADER}"
)
```
If that is correct where does `VENDOR_CLI_HEADER` get set?  I guess my point is that something needs to be specified by the user doing the compilation and if thats the case then shouldn't it be in the README?